File: app/src/main/java/hpn332/cb/ui/activity/EditActivity.java
package hpn332.cb.ui.activity;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import hpn332.cb.R;
import hpn332.cb.ui.fragment.AddFragmentBacklog;
import hpn332.cb.ui.fragment.AddFragmentProject;
import hpn332.cb.ui.fragment.AddFragmentTag;
import hpn332.cb.ui.fragment.AddFragmentTask;
import hpn332.cb.ui.fragment.DialogFragmentColorPicker;
import hpn332.cb.ui.fragment.EditFragmentBacklog;
import hpn332.cb.ui.fragment.EditFragmentProject;
import hpn332.cb.ui.fragment.EditFragmentTag;
import hpn332.cb.ui.fragment.EditFragmentTask;
import hpn332.cb.ui.fragment.ListFragmentBacklog;
import hpn332.cb.utils.Key;
import hpn332.cb.utils.Type;
import hpn332.library.view.ColorPanelView;

public class EditActivity extends AppCompatActivity
		implements DialogFragmentColorPicker.ColorPickerDialogListener,
		ListFragmentBacklog.OnBacklogFragment {

	private static final String TAG = "EditActivity";

	public static ColorPanelView color_panel;

	public static int backlogId = 0;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_edit);
		Log.d(TAG, "onCreate: start");

		init();

		Log.d(TAG, "onCreate: end");
	}

	private void init() {
		Log.d(TAG, "init: start");

		setFragment(checkTypeAndGetFragment());

		Log.d(TAG, "init: end");
	}

	private void setFragment(Fragment fragment) {
		Log.d(TAG, "setFragment: start");

		getSupportFragmentManager()
				.beginTransaction()
				.add(R.id.frameLayout, fragment)
				.commit();

		Log.d(TAG, "setFragment: end");
	}

	@Override
	public void onColorSelected(int color) {
		color_panel.setColor(color);
		Log.d(TAG, "onColorSelected: color " + color);
	}

	@Override
	public void onDialogDismissed(int color) {
		color_panel.setColor(color);
		Log.d(TAG, "onDialogDismissed: color " + color);
	}

	protected Fragment checkTypeAndGetFragment() {
		Log.d(TAG, "checkTypeAndGetFragment: start type is :: ");

		switch (getIntent().getIntExtra(Key.TYPE, 0)) {
			case Type.ADD_PROJECT:
				return new AddFragmentProject();
			case Type.ADD_BACKLOG:
				return new AddFragmentBacklog();
			case Type.ADD_TASK:
				return new AddFragmentTask();
			case Type.ADD_TAG:
				return new AddFragmentTag();
			case Type.EDIT_PROJECT:
				return new EditFragmentProject();
			case Type.EDIT_BACKLOG:
				return new EditFragmentBacklog();
			case Type.EDIT_TASK:
				return new EditFragmentTask();
			case Type.EDIT_TAG:
				return new EditFragmentTag();
		}

		throw new IllegalArgumentException("Unknown type");
	}

	@Override
	public void onClickBacklog(int backlogId) {
		Log.d(TAG, "onClickBacklog: backlog id :: " + backlogId);

		EditActivity.backlogId = backlogId;
	}
}

File: app/src/main/java/hpn332/cb/view/activity/EditActivity.kt
package hpn332.cb.view.activity

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.util.Log

import hpn332.cb.R
import hpn332.cb.utils.presenter.Backlog
import hpn332.cb.view.fragment.AddFragmentBacklog
import hpn332.cb.view.fragment.AddFragmentProject
import hpn332.cb.view.fragment.AddFragmentTag
import hpn332.cb.view.fragment.AddFragmentTask
import hpn332.cb.view.fragment.DialogFragmentColorPicker
import hpn332.cb.view.fragment.EditFragmentBacklog
import hpn332.cb.view.fragment.EditFragmentProject
import hpn332.cb.view.fragment.EditFragmentTag
import hpn332.cb.view.fragment.EditFragmentTask
import hpn332.cb.utils.Key
import hpn332.cb.utils.Type
import hpn332.library.view.ColorPanelView

class EditActivity : AppCompatActivity(), DialogFragmentColorPicker.ColorPickerDialogListener, Backlog {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_edit)
		Log.d(TAG, "onCreate: start")

		init()

		Log.d(TAG, "onCreate: end")
	}

	private fun init() {
		Log.d(TAG, "init: start")

		setFragment(checkTypeAndGetFragment())

		Log.d(TAG, "init: end")
	}

	private fun setFragment(fragment: Fragment) {
		Log.d(TAG, "setFragment: start")

		supportFragmentManager
				.beginTransaction()
				.add(R.id.frameLayout, fragment)
				.commit()

		Log.d(TAG, "setFragment: end")
	}

	override fun onColorSelected(color: Int) {
		color_panel!!.color = color
		Log.d(TAG, "onColorSelected: color " + color)
	}

	override fun onDialogDismissed(color: Int) {
		color_panel!!.color = color
		Log.d(TAG, "onDialogDismissed: color " + color)
	}

	private fun checkTypeAndGetFragment(): Fragment {
		Log.d(TAG, "checkTypeAndGetFragment: start type is :: ")

		when (intent.getIntExtra(Key.TYPE, 0)) {
			Type.ADD_PROJECT  -> return AddFragmentProject()
			Type.ADD_BACKLOG  -> return AddFragmentBacklog()
			Type.ADD_TASK     -> return AddFragmentTask()
			Type.ADD_TAG      -> return AddFragmentTag()
			Type.EDIT_PROJECT -> return EditFragmentProject()
			Type.EDIT_BACKLOG -> return EditFragmentBacklog()
			Type.EDIT_TASK    -> return EditFragmentTask()
			Type.EDIT_TAG     -> return EditFragmentTag()
		}

		throw IllegalArgumentException("Unknown type")
	}

	override fun onClickBacklog(backlogId: Int) {
		Log.d(TAG, "onClickBacklog: backlog id :: " + backlogId)

		EditActivity.backlogId = backlogId
	}

	companion object {

		private val TAG = "EditActivity"

		var color_panel: ColorPanelView? = null

		var backlogId = 0
	}
}
